framework/src/base/MooseVariableScalar.C

```diff
@@ -23,6 +23,9 @@
 #include "libmesh/numeric_vector.h"
 #include "libmesh/dof_map.h"
 
+// C++
```

Comment:
```
roystgnr@leela:(roys_junk3) ~/git/moose-fresh
$ egrep -i '// (system|C\+\+)' framework/include/*/*.h | wc
     35     118    2112
roystgnr@leela:(roys_junk3) ~/git/moose-fresh
$ egrep -i '// libMesh includes' framework/include/*/*.h | wc
     70     210    4668
```

Should I get the other hundred instances while I'm at it?  That should probably be a separate PR.

I'd rather not remove any of them at all.  When I was removing unnecessary header dependencies, I discovered that dependency-tracking best practices (e.g. https://stackoverflow.com/a/2762596 ) were followed less often than not.  I'd like to have *some* comments that express "This isn't random order; someone put these headers below those headers for a reason, and here's what category they were sorted by".